app/views/spree/orders/form/_update_buttons.html.haml

```diff
@@ -1,7 +1,7 @@
 .row
   .columns.small-12.medium-3
-    - if current_order.andand.distributor == @order.distributor
```

Comment:
Yes, I see. The problem is that when we tested this `current_order` was returning nil *hitting https://github.com/coopdevs/spree/blob/spree-upgrade-step-6/core/lib/spree/core/controller_helpers/order.rb#L18) thus not showing the button right after purchasing.

Knowing the context you provided we can investigate where is it that the behaviour breaks. Perhaps we find a more reliable way to implement it.